aptos-move/framework/move-stdlib/src/natives/vector.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Implementation of native functions for utf8 strings.
+
+use aptos_gas_schedule::gas_params::natives::move_stdlib::{
+    VECTOR_RANGE_MOVE_BASE, VECTOR_RANGE_MOVE_PER_INDEX_MOVED,
+};
+use aptos_native_interface::{
+    safely_pop_arg, RawSafeNative, SafeNativeBuilder, SafeNativeContext, SafeNativeError,
+    SafeNativeResult,
+};
+use move_core_types::gas_algebra::NumArgs;
+use move_vm_runtime::native_functions::NativeFunction;
+use move_vm_types::{
+    loaded_data::runtime_types::Type,
+    values::{Value, VectorRef},
+};
+use smallvec::{smallvec, SmallVec};
+use std::collections::VecDeque;
+
+use super::mem::get_feature_not_available_error;
+
+/// The generic type supplied to aggregator snapshots is not supported.
+pub const EINDEX_OUT_OF_BOUNDS: u64 = 0x03_0001;
```

Comment:
there is this utility now  - https://github.com/aptos-labs/aptos-core/blob/main/types/src/error.rs

I'll use that, but it might be worth revisiting where to put it, to not require importing whole aptos-types for that.